CMake/UseITK.cmake

```diff
@@ -228,8 +239,13 @@ foreach(_factory_name ${ITK_FACTORY_LIST})
 endforeach()
 #-----------------------------------------------------------------------------
 if(NOT ITK_NO_IO_FACTORY_REGISTER_MANAGER)
-
+  #FIXME this only sets properties under the ITK-build/wrapping subdirectory
   set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ITK_IO_FACTORY_REGISTER_MANAGER)
   include_directories(BEFORE ${CMAKE_CURRENT_BINARY_DIR}/ITKFactoryRegistration)
+endif()
 
+if(NOT ITK_NO_FFT_FACTORY_REGISTER_MANAGER)
+  #FIXME this only sets properties under the ITK-build/wrapping subdirectory
+  set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ITK_FFT_FACTORY_REGISTER_MANAGER)
+  include_directories(BEFORE ${CMAKE_CURRENT_BINARY_DIR}/ITKFactoryRegistration)
 endif()
```

Comment:
Some issues with this block:
- `UseITK.cmake` seems to be called only in relation to `ITK/Wrapping`, which limits the scope of the compiler definitions and include directories it's trying to set. I had to reproduce these lines in `CMakeLists.txt` files under ITKFFT to get default factories loaded in for FFT tests.
- Is it correct behavior that `UseITK.cmake` is only called in relation to wrapping? I need to look more into how/whether default factories are processed when wrapping is turned off. It seems like wrapping and setting up default factories should be independent operations, though.
- Under what context would we want to disable factory register managers?
- Is it intended that these properties are set on the parent folder instead of the Wrapping subdirectory?

Reply:
`UseITK.cmake` is invoked by library users. And the library is implicitly used during the compilation of the wrappings.

Reply:
Gotcha. In that case test drivers should also invoke `UseITK.cmake`, correct?

Reply:
From testing it does not look like `UseITK.cmake` is used at all inside ITK unless wrappings are compiled. Is the factory register manager used solely for external projects? It doesn't look like factory managers are being used for initialized factory lists at all; for instance, `itkTransformFactoryBase` is [manually initializing most transform factories](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/TransformFactory/src/itkTransformFactoryBaseRegister.hxx#L104) when initialized with [`GetFactory`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/TransformFactory/src/itkTransformFactoryBase.cxx#L96).

Reply:
I guess we should have that factory manager, and it should register all the compiled classes. It is made available to user code, so users can add their own implementation of classes which are created through the factory.